src/coreclr/jit/hwintrinsicarm64.cpp

```diff
@@ -1800,27 +1800,10 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
                 }
             }
 
-            GenTree* loadIntrinsic = gtNewSimdHWIntrinsicNode(retType, op1, intrinsic, simdBaseJitType, simdSize);
-            // This operation contains an implicit indirection
-            //   it could point into the global heap or
-            //   it could throw a null reference exception.
-            //
-            loadIntrinsic->gtFlags |= (GTF_GLOB_REF | GTF_EXCEPT);
-
             assert(HWIntrinsicInfo::IsMultiReg(intrinsic));
 
-            const unsigned lclNum = lvaGrabTemp(true DEBUGARG("Return value temp for multireg intrinsic"));
-            impAssignTempGen(lclNum, loadIntrinsic, sig->retTypeSigClass, CHECK_SPILL_ALL);
-
-            LclVarDsc* varDsc = lvaGetDesc(lclNum);
-            // The following is to exclude the fields of the local to have SSA.
-            varDsc->lvIsMultiRegRet = true;
-
-            GenTreeLclVar* lclVar = gtNewLclvNode(lclNum, varDsc->lvType);
-            lclVar->SetDoNotCSE();
-            lclVar->SetMultiReg();
-
-            retNode = lclVar;
+            op1     = gtNewSimdHWIntrinsicNode(retType, op1, intrinsic, simdBaseJitType, simdSize);
+            retNode = impAssignMultiRegTypeToVar(op1, sig->retTypeSigClass DEBUGARG(CorInfoCallConvExtension::Managed));
```

Comment:
`impAssignMultiRegTypeToVar()` method doesn't set `varDsc->lvIsMultiRegRet = true;`

See https://github.com/dotnet/runtime/pull/66551/commits/65c0af67b66c71989e62c5d653339d1a51982347

Reply:
I see it does...?

https://github.com/dotnet/runtime/blob/8f06013212ebc2813948f78ad864ebf101047e0b/src/coreclr/jit/importer.cpp#L17728-L17743

It's even redundant, technically, `impAssignStructPtr` will set it when called from `impAssignTempGen`.

Reply:
ah, I was referring to the PR fork that was not doing it. Never mind.